lib/common/xml.c

```diff
@@ -460,6 +460,31 @@ pcmk__xe_first_child(const xmlNode *parent, const char *node_name,
     return NULL;
 }
 
+/*!
+ * \internal
+ * \brief Copy XML attributes, expanding \c ++ and \c +=
+ *
+ * This is similar to \c xmlCopyPropList() except that \c ++ and \c += are
+ * expanded where appropriate. See \c expand_plus_plus() for details.
+ *
+ * \param[in,out] target  XML element to receive copied attributes from \p src
+ * \param[in]     src     XML element whose attributes to copy to \p target
+ */
+void
+pcmk__xe_copy_attrs(xmlNode *target, const xmlNode *src)
+{
+    CRM_CHECK((src != NULL) && (target != NULL), return);
```

Comment:
> Have you double-checked that callers can never pass NULL?

Effectively yes. The one caveat is that `cib_perform_op()`'s checking is not great. It null-checks `current_cib` but not `*current_cib`, except that in some cases it will try to dereference `*current_cib`...

`cib_perform_op()` calls `pcmk__xe_copy_attrs()` both directly and via `cib_process_erase()`, using `*current_cib`.

Reply:
> xml_element.c

https://projects.clusterlabs.org/T815